cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/runtimejob/DataprocRuntimeJobManager.java

```diff
@@ -418,7 +419,23 @@ private boolean isDeleteLifecycleEnabled(String bucketName) {
       }
       if (rule.getAction() instanceof BucketInfo.LifecycleRule.DeleteLifecycleAction &&
         rule.getCondition().getDaysSinceCustomTime() > 0) {
-        return true;
+        if (!provisionerContext.getProperties().containsKey(DataprocUtils.ARTIFACTS_COMPUTE_HASH_TIME_BUCKET_DAYS)) {
+          LOG.warn("ArtifactsHashTimeBucket property not set for {}, ignoring check for it's value being less than " +
```

Comment:
seems like we don't expect this to ever happen right? would be good to add a comment describing if this is ever possible

Reply:
Not really sure when this case could happen. Added a check as a fail-safe in case this property gets deleted in DistributedProgramRunner.